edu.gemini.lch.services/src/main/java/edu/gemini/lch/services/ModelFactory.java

```diff
@@ -20,22 +22,17 @@
 
     /**
      * Creates a laser night for the given date (in local time of the site).
-     * @param site
-     * @param day
-     * @return
      */
-    public static LaserNight createNight(Site site, DateTime day) {
-        SunRiseSet sunCalc = createSunCalculator(site, day.withTimeAtStartOfDay());
-        DateTime sunsetUTC  = new DateTime(sunCalc.getSunset(),  DateTimeZone.UTC);
-        DateTime sunriseUTC = new DateTime(sunCalc.getSunrise(), DateTimeZone.UTC);
+    public static LaserNight createNight(Site site, ZonedDateTime day) {
+        SunRiseSet sunCalc = createSunCalculator(site, day.toLocalDate().atStartOfDay(site.getZoneId()));
+        ZonedDateTime sunsetUTC  = ZonedDateTime.ofInstant(sunCalc.getSunset().toInstant(), ZoneId.systemDefault());
```

Comment:
Why do `ofInstant(x).toInstant`? It seems you could just use `x`?

Reply:
This is a hideous example of Java 8 time at work.

1. We are working with a `SunRiseSet` as created by `ModelFactory.createSunCalculator`, which is from JSky, which uses `Date`.
2. We want a `LaserNight`, which takes `ZonedDateTime`s.
3. I think the way to get a `ZonedDateTime` from a `Date` is to through an `Instant`, so we get this convoluted mess. So we use `toInstant` to turn the `Date` objects into `Instant`s, and then the `ofInstant` to turn the `Instant` objects into `ZonedDateTime`s.

Digging up all this elegance in conversion is what led to this task taking as long as it has. It's like the Monty Python Department of Redundancy Department.